components/script/dom/bindings/str.rs

```diff
@@ -113,6 +113,74 @@ pub fn is_token(s: &[u8]) -> bool {
     })
 }
 
+/// Returns whether the language is matched, as defined by
+/// [RFC 4647](https://tools.ietf.org/html/rfc4647#section-3.3.2).
+pub fn extended_filtering(tag: &str, range: &str) -> bool {
+    let range_tags: Vec<&str> = range.split(',').collect();
+    let mut is_match: bool = true;
+
+    for range_tag in range_tags {
+        // step 1
+        let css_tags: Vec<&str> = range_tag.split('\x2d').collect();
+        let elem_tags: Vec<&str> = tag.split('\x2d').collect();
```

Comment:
Let's not introduce the `css` and `elem` terms here. The extended filtering algorithm does not define these at all. Instead, name them `range_subtags` and `tag_subtags`.